packages/headless/src/features/commerce/product-templates/product-templates-manager.ts

```diff
@@ -0,0 +1,40 @@
+import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
+import {
+  buildTemplatesManager,
+  Template,
+  TemplateCondition,
+} from '../../templates/templates-manager';
+
+export type ProductTemplate<Content = unknown> = Template<
+  ProductRecommendation,
+  Content
+>;
+export type ProductTemplateCondition = TemplateCondition<ProductRecommendation>;
+
+export interface ProductTemplatesManager<Content = unknown> {
+  /**
+   * Registers any number of result templates in the manager.
+   * @param templates (...ResultTemplate<Content>) A list of templates to register.
```

Comment:
```suggestion
   * @param templates (...Template<Product, Content>) A list of templates to register.
```

(I'm preemptively suggesting Product as the documented type since this is about to change anyways).